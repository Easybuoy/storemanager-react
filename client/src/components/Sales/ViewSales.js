import React, { Component } from 'react'
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { toast } from 'react-toastify';

import { viewSales } from '../../actions/saleActions';
import Loading from '../Common/Loading';

 export class ViewSales extends Component {
    componentDidMount(){
         this.props.viewSales();
     }

    render() {
        if (Object.keys(this.props.errors).length > 0) {

            toast.error(this.props.errors.message);
          }

          const { sales, loading } = this.props.sales;

       if (loading) { 
        return (
          <div>
          <Loading />
          </div>
        )
      }

        if (sales.length === 0) {
            // return no sale found.
            return(
                <div id="main">
                    <div className="topmargin">
                        <h1 className="text-center">No Sales Have Been Made.</h1>
                    </div>
                </div>  
            )
        }
    return (
        
      <div id="main">
      <section className="topmargin">
                <div className="container text-center">
                        <table className="table">
                                <caption>Sales</caption>
                                <thead>
                                  <tr>
                                    <th scope="col">Date</th>
                                    <th scope="col">Product</th>
                                    <th scope="col">Amount</th>
                                    <th scope="col">Quantity</th>
                                  </tr>
                                </thead>
                                <tbody>
                                    {
                                        sales.map((sale, key) => {
                                            return (
                                <tr key={key}>
                                    <td data-label="Date">{sale.created_at}</td>
                                    <td data-label="Product">{sale.name}</td>
                                    <td data-label="Amount">${sale.price}</td>
                                    <td data-label="Quantity">{sale.quantity}</td>
                                </tr>
                                            )
                                        })
                                    }
                                </tbody>
                              </table>
                </div>
            </section>    

      </div>
    )
  }
}

ViewSales.propTypes = {
    sales: PropTypes.object.isRequired,
    errors: PropTypes.object.isRequired,
    viewSales: PropTypes.func.isRequired
}

const mapStateToProps = state => ({
    sales: state.sales,
    errors: state.errors
});

export default connect(mapStateToProps, { viewSales })(ViewSales);