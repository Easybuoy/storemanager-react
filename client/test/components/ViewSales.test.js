import React from 'react';
import { shallow } from 'enzyme';

import ViewSales from '../../src/components/Sales/ViewSales';

describe('<ViewSales />', () => {
    it('renders the ViewSales component correctly', () => {
        const wrapper = shallow(<ViewSales />);
        expect(wrapper).toMatchSnapshot();
    });
});