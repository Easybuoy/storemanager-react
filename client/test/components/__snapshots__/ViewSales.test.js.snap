// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ViewSales /> renders the ViewSales component correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ViewSales
    errors={
      Object {
        "addMatchers": [Function],
        "advanceTimersByTime": [Function],
        "autoMockOff": [Function],
        "autoMockOn": [Function],
        "clearAllMocks": [Function],
        "clearAllTimers": [Function],
        "deepUnmock": [Function],
        "disableAutomock": [Function],
        "doMock": [Function],
        "dontMock": [Function],
        "enableAutomock": [Function],
        "fn": [Function],
        "genMockFromModule": [Function],
        "isMockFunction": [Function],
        "mock": [Function],
        "requireActual": [Function],
        "requireMock": [Function],
        "resetAllMocks": [Function],
        "resetModuleRegistry": [Function],
        "resetModules": [Function],
        "restoreAllMocks": [Function],
        "retryTimes": [Function],
        "runAllImmediates": [Function],
        "runAllTicks": [Function],
        "runAllTimers": [Function],
        "runOnlyPendingTimers": [Function],
        "runTimersToTime": [Function],
        "setMock": [Function],
        "setTimeout": [Function],
        "spyOn": [Function],
        "unmock": [Function],
        "useFakeTimers": [Function],
        "useRealTimers": [Function],
      }
    }
    sales={
      Object {
        "sales": Array [],
      }
    }
    viewSales={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="topmargin"
      >
        <h1
          className="text-center"
        >
          No Sales Have Been Made.
        </h1>
      </div>,
      "id": "main",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h1
          className="text-center"
        >
          No Sales Have Been Made.
        </h1>,
        "className": "topmargin",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "No Sales Have Been Made.",
          "className": "text-center",
        },
        "ref": null,
        "rendered": "No Sales Have Been Made.",
        "type": "h1",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="topmargin"
        >
          <h1
            className="text-center"
          >
            No Sales Have Been Made.
          </h1>
        </div>,
        "id": "main",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h1
            className="text-center"
          >
            No Sales Have Been Made.
          </h1>,
          "className": "topmargin",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "No Sales Have Been Made.",
            "className": "text-center",
          },
          "ref": null,
          "rendered": "No Sales Have Been Made.",
          "type": "h1",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
